app/src/main/java/edu/osu/pcv/marslogger/IMUManager.java

```diff
@@ -207,7 +230,25 @@ public final void onSensorChanged(SensorEvent event) {
             SensorPacket syncedData = syncInertialData();
             if (syncedData != null && mRecordingInertialData) {
                 try {
-                    mDataWriter.write(syncedData.toString() + "\n");
+                    if (currLocation != null) {
```

Comment:
Depending on the target application, it may be inconvenient to log the IMU data along with the GPS data, so I would recommend log the IMU data and GPS data separately. For instance IMUManager to log IMU data, a GPS manager to log GPS data.
This way the functionalities are separated into two independent classes. And it also makes the line format more amenable to a data previewer/loader, e.g., matlab readmatrix.

Reply:
With brief testing, I found several minor issues. 1. To compile the app, we need to include include ':LocationProvider' in the root settings.gradle file. 2. The accelerometer z axis value appears twice in each IMU data line. I will keep working on it in the coming days.

Reply:
Sorry for the delay. I got swamped with work. Hopefully, I can work on it starting next week. Rest assured I will deal with all the points you brought up here as I think they are valid.

Reply:
I fixed the issues you were facing and also created a new class called GPSManager which is used to save the GPS values in a separate csv file (the timestamps are the same as used for the IMU values so as to keep both IMU and GPS values synchronized).

Reply:
I don't see that class GPSManager in your fork yet. I think you can save the LocationResult in the LocationCallback structure, Specifically, create a file stream, then write to the stream the Location obtained in LocationCallback, then close the stream when the thread quits, as done for the IMU data in IMUManager.